### PR TITLE
Added new preferences dialog to allow dynamic changing of time zone.

### PR DESCRIPTION
**Pull request by michael-gebis**
- The main change in this pull request is the ability to set the time zone through a new "set-preferences" button in the web interface
- The preferences are saved in non-volatile memory, and are retained across power cycling.
- The time zone may now be a fractional hour offset
- Setting and unsetting daylight saving time is now a single checkbox
- Added a workaround to an issue where the hostname may not be set (at least in the latest non-beta arduino ESP32 libraries (V2.0.14))
- Updated README.md to include info on the above changes

### CODE REVIEW

**Comment by michael-gebis** [comment]
Harshit, thank you for your time on the original project, and also any time you spend reviewing my changes.  One other request: I noticed this repo does not have a LICENSE file.  I would like to ask you to please add one--hopefully something like the MIT license, but anything open source would be awesome.  Thanks!

File: wandering-hour-clock.ino [hunk 1]
```diff
@@ -17,14 +17,25 @@
...
 
 // Replace the ssid and password in secrets.h
 const char* ssid = SECRET_SSID;
 const char* password = SECRET_PASSWORD;
 
+// Hostname
+const char* hostname = "wander";
+
+// Preferences library namespace and keys
+const char* pref_namespace = "whc"; // "Wandering Hour Clock"
```
Comment (sanghviharshit) [inline wandering-hour-clock.ino:35]: can we change this to `wandering-hour-clock` to keep it backward compatible with existing installs?

**Review by sanghviharshit — CHANGES_REQUESTED** [review]
Wow! This is everything I have been hoping to add to this project. Thank you for the PR! Looks great except the small change for the hostname.

File: wandering-hour-clock.ino (hunk 1 above)
Comment (michael-gebis) [reply wandering-hour-clock.ino:35]: Thanks for reviewing everything!  I'll make that change now!

**Pull request merged**